src/lib/profanity-filter.ts

```
@@ -218,12 +222,49 @@ export class ProfanityFilter {
     return result.hasProfanity;
   }
 
+  private normalizeVietnamese(text: string): string {
+    let normalized = text.normalize("NFC").toLowerCase().trim();
+    normalized = normalized.replace(/[\u200B-\u200D\uFEFF\u00AD]/g, "");
```

Biome (1.9.4)

[error] A character class cannot match a joined character sequence.

A zero width joiner composes several emojis into a new one. Replace the character class with an alternation.

(lint/suspicious/noMisleadingCharacterClass)